script/hassfest/config_flow.py

```diff
@@ -32,12 +32,22 @@ def validate_integration(config: Config, integration: Integration):
 
     config_flow = config_flow_file.read_text()
 
-    needs_unique_id = integration.domain not in UNIQUE_ID_IGNORE and (
-        "async_step_hassio" in config_flow
-        or any(
-            bool(integration.manifest.get(key))
-            for keys in DISCOVERY_INTEGRATIONS.values()
-            for key in keys
+    needs_unique_id = (
+        integration.domain not in UNIQUE_ID_IGNORE
+        and (
+            "async_step_hassio" in config_flow
+            or any(
+                bool(integration.manifest.get(key))
+                for keys in DISCOVERY_INTEGRATIONS.values()
+                for key in keys
+            )
```

Comment:
We no longer need to check this, since each discovery step will now have a default.